test_stage_2.py

```diff
@@ -183,39 +191,50 @@ def handle_single(ref_image_path,pose_video_path):
             width,
             height,
             len(pose_list),
-            args.steps,
-            args.cfg,
+            steps,
+            cfg,
             generator=generator,
-            context_frames=args.S,
+            context_frames=slice_num,
             context_stride=1,
-            context_overlap=args.O,
+            context_overlap=overlap,
         ).videos
 
 
         m1 = config.pose_guider_path.split('.')[0].split('/')[-1]
         m2 = config.motion_module_path.split('.')[0].split('/')[-1]
 
-        save_dir_name = f"{time_str}-{args.cfg}-{m1}-{m2}"
+        save_dir_name = f"{time_str}-{cfg}-{m1}-{m2}"
         save_dir = Path(f"./output/video-{date_str}/{save_dir_name}")
         save_dir.mkdir(exist_ok=True, parents=True)
 
         result = scale_video(video[:,:,:L], original_width, original_height)
+        output_path1 = f"{save_dir}/{ref_name}_{pose_name}_{cfg}_{steps}_{skip}.mp4"
         save_videos_grid(
             result,
-            f"{save_dir}/{ref_name}_{pose_name}_{args.cfg}_{args.steps}_{args.skip}.mp4",
+            output_path1,
             n_rows=1,
-            fps=src_fps if args.fps is None else args.fps,
-        )    
-
-        video = torch.cat([ref_image_tensor, pose_tensor[:,:,:L], video[:,:,:L]], dim=0) 
-        video = scale_video(video, original_width, original_height)     
-        save_videos_grid(
-            video,
-            f"{save_dir}/{ref_name}_{pose_name}_{args.cfg}_{args.steps}_{args.skip}_{m1}_{m2}.mp4",
-            n_rows=3,
-            fps=src_fps if args.fps is None else args.fps,
+            fps=src_fps if fps is None else fps,
         )
 
+        # print("ref_image_tensor size:", ref_image_tensor.size())
+        # print("pose_tensor size:", pose_tensor[:, :, :L].size())
+        # print("video size:", video[:, :, :L].size())
+
+        # video = torch.cat([ref_image_tensor, pose_tensor[:,:,:L], video[:,:,:L]], dim=0) 
+        # video = scale_video(video, original_width, original_height)     
+        # output_path2 = f"{save_dir}/{ref_name}_{pose_name}_{cfg}_{steps}_{skip}_{m1}_{m2}.mp4"
+        # save_videos_grid(
+        #     video,
+        #     output_path2,
+        #     n_rows=3,
+        #     fps=src_fps if fps is None else fps,
+        # )
```

Comment:
I commented this out, because I was consistently getting the following error:

```
Traceback (most recent call last):
  File "/root/test_stage_2.py", line 238, in <module>
    main()
  File "/root/test_stage_2.py", line 232, in main
    handle_single(ref_image_path, pose_video_path) 
  File "/root/test_stage_2.py", line 211, in handle_single
    video = torch.cat([ref_image_tensor, pose_tensor[:,:,:L], video[:,:,:L]], dim=0) 
RuntimeError: Sizes of tensors must match except in dimension 0. Expected size 540 but got size 536 for tensor number 2 in the list.
```

The way the code was saving outputs to the disk before, these errors would fail silently. But with this PR, they will be blocking.

Reply:
I just realized the cause: This is for video generations that are 16:9 aspect ratio, e.g. 960 height and 540 width. 540 converted to latent space is halfway between 536 and 544...

Reply:
Solved by choosing a different height and width